src/interface/TWDerivationPath.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include <TrustWalletCore/TWDerivationPath.h>
+#include <TrustWalletCore/TWPurpose.h>
+#include <TrustWalletCore/TWString.h>
+
+#include "../DerivationPath.h"
+
+using namespace TW;
+
+struct TWDerivationPath* _Nonnull TWDerivationPathCreate(enum TWPurpose purpose, uint32_t coin, uint32_t account, uint32_t change, uint32_t address) {
+    return new TWDerivationPath{DerivationPath(purpose, coin, account, change, address)};
+}
+
+struct TWDerivationPath* _Nullable TWDerivationPathCreateWithString(TWString* _Nonnull string) {
+    auto& str = *reinterpret_cast<const std::string*>(string);
+    try {
+        return new TWDerivationPath{DerivationPath(str)};
+    } catch (...) {
+        return nullptr;
+    }
+}
+
+void TWDerivationPathDelete(struct TWDerivationPath* _Nonnull path) {
+    delete path;
```

Comment:
Perhaps delete it's indices as well?

Reply:
Indices are not pointer, destructor will be called automatically